kube/file-vault/file-vault-deployment.yml

```diff
@@ -0,0 +1,261 @@
+              valueFrom:
+                secretKeyRef:
+                  name: keycloak-client
+                  key: id
+            - name: PROXY_LISTEN
+              value: 127.0.0.1:3001
+            - name: PROXY_UPSTREAM_URL
+              value: "https://127.0.0.1:3000"
+            - name: PROXY_REDIRECTION_URL
+            {{ if eq .KUBE_NAMESPACE .PROD_ENV }}
+              value: https://fv.prod.{{ .APP_NAME }}.homeoffice.gov.uk
+            {{ else if eq .KUBE_NAMESPACE .STG_ENV }}
+              value: https://fv-{{ .APP_NAME }}.stg.sas.homeoffice.gov.uk
+            {{ else if eq .KUBE_NAMESPACE .UAT_ENV }}
+              value: https://fv-{{ .APP_NAME }}.uat.sas-notprod.homeoffice.gov.uk
+            {{ else if eq .KUBE_NAMESPACE .BRANCH_ENV }}
+              value: https://fv-{{ .DRONE_SOURCE_BRANCH }}.branch.sas-notprod.homeoffice.gov.uk
+            {{ end }}
+            - name: PROXY_DISCOVERY_URL
+            {{ if eq .KUBE_NAMESPACE .PROD_ENV }}
+              value: https://acp-sso.prod.acp.homeoffice.gov.uk/realms/fmr
+            {{ else if eq .KUBE_NAMESPACE .STG_ENV }}
+              value: https://acp-sso.prod.acp.homeoffice.gov.uk/realms/fmr
+            {{ else if eq .KUBE_NAMESPACE .UAT_ENV }}
+              value: https://acp-sso.notprod.acp.homeoffice.gov.uk/realms/fmr-notprod
+            {{ else if eq .KUBE_NAMESPACE .BRANCH_ENV }}
+              value: https://acp-sso.notprod.acp.homeoffice.gov.uk/realms/fmr-notprod
+            {{ end }}
+          args:
+            - --listen=127.0.0.1:3001
+            # the url which is used to retrieve the OpenID configuration
+            - --discovery-url=$(PROXY_DISCOVERY_URL)
+            # the endpoint where requests are proxied to
+            - --upstream-url=https://127.0.0.1:3000
+            # URls that you wish to protect.
+            - --resources=uri=/*
+            - --client-id=$(PROXY_CLIENT_ID)
+            - --client-secret=$(PROXY_CLIENT_SECRET)
+            - --tls-cert=/certs/tls.crt
+            - --tls-private-key=/certs/tls.key
+            - --enable-default-deny=false
+            - --http-only-cookie=true
+            - --enable-security-filter=true
+            - --enable-request-id=true
+            - --enable-logging=true
+            - --enable-json-logging=true
+            - --redirection-url=$(PROXY_REDIRECTION_URL)
+            - --tls-min-version=tlsv1.2
+            {{ if eq .KUBE_NAMESPACE .BRANCH_ENV }}
+            - --verbose=true
+            {{ end }}
+          ports:
+            - containerPort: 10080
+            - containerPort: 10443
+          securityContext:
+            runAsNonRoot: true
+          terminationMessagePath: /dev/termination-log
+          terminationMessagePolicy: File
+          volumeMounts:
+            - mountPath: /certs
+              name: certs
+            - mountPath: /etc/ssl/certs
+              name: bundle
+              readOnly: true
+
+        - name: nginx-proxy
```

Comment:
even though nginx-proxy listens ports 10443 and 10080 by default 
it worth adding them explicitly to avoid the confusions
```
 ports:
            - containerPort: 10080
            - containerPort: 10443
 ```

Reply:
Added